### PR TITLE
Adding Mattermost-supported plugins to product docs

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/docs/issues/7005

GitHub repository README updates to de-duplicate content is managed via separate PRs:
- GitHub: https://github.com/mattermost/mattermost-plugin-github/pull/777
- GitLab: https://github.com/mattermost/mattermost-plugin-gitlab/pull/494
- Jira: https://github.com/mattermost/mattermost-plugin-jira/pull/1076
- SN: https://github.com/mattermost/mattermost-plugin-servicenow/pull/194
- Zoom: https://github.com/mattermost/mattermost-plugin-zoom/pull/376
- MS Teams: https://github.com/mattermost/mattermost-plugin-msteams/pull/645
- AI: https://github.com/mattermost/mattermost-plugin-ai/pull/178
- Metrics plugin: https://github.com/mattermost/mattermost-plugin-metrics/pull/26
- Channel Export: https://github.com/mattermost/mattermost-plugin-channel-export/pull/42

Separate PRs are being created to manage all in-product docs links currently taking users to GH READMEs, and take them to product docs instead.

The Google Calendar plugin documentation has been migrated from the Mattermost Product Documentation to its GitHub repository README via: https://github.com/mattermost/mattermost-plugin-google-calendar/pull/74

Note: Legal Hold config docs & the new v2 Survey Plugin docs will be consolidated into the Mattermost Product Documentation in a future iteration. The existing User Satisfaction Survey docs will also be reviewed and aligned in a future iteration.

### CODE REVIEW

File: source/guides/integrate-and-extend.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Your Mattermost deployment comes with the following integrations you can configure and use:
+
+* :doc:`AI Copilot plugin </integrate/ai-copilot-plugin>` - Connect and use the Mattermost AI Copilot integration.
+* :doc:`Channel Export plugin </integrate/export-mattermost-channel-data>` - Learn how to export channel data.
+* :doc:`GitHub interoperability </integrate/github-interoperability>` - Connect your GitHub instance to your Mattermost instance.
+* :doc:`GitLab interoperability </integrate/gitlab-interoperability>` - Connect your GitLab instance to your Mattermost instance.
+* :doc:`Jira interoperability </integrate/jira-interoperability>` - Connect your Jira instance to your Mattermost instance.
+* :doc:`Legal hold
+* :doc:`Performance metrics
```
Comment (hanzei) [inline source/guides/integrate-and-extend.rst:33]: I wonder if we should move the Metrics plugin out of this category. It's not an integration, but more of a tool. It's more like "Calls" and less like the "GitHub" plugin.

File: source/guides/integrate-and-extend.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    Zoom </integrate/zoom-interoperability>
+
+Mattermost offers many ways to add product functionality, custom-tailor your workflows, and connect Mattermost to your mission-critical tools. 
+
+Interoperability with pre-packaged integrations
+-----------------------------------------------
+
+Your Mattermost deployment comes with the following integrations you can configure and use:
+
+* :doc:`AI Copilot plugin </integrate/ai-copilot-plugin>` - Connect and use the Mattermost AI Copilot integration.
```
Comment (hanzei) [inline source/guides/integrate-and-extend.rst:27]: Do we need an `Alpha` tag?

File: source/configure/plugins-configuration-settings.rst [hunk 3]
```diff
@@ -1619,8 +1423,8 @@ In addition to the Mattermost plugin signing key built into the server, each pub
 | This feature's ``config.json`` setting is ``"SignaturePublicKeyFiles": {}`` with string array input consisting of contents that are relative or absolute paths to signature files.              |
 +-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
-.. config:setting:: plugins-zoomchimeraoauthproxyurl
-  :displayname: Chimera OAuth proxy URL (Plugins - Zoom)
+.. config:setting:: plugins-chimeraoauthproxyurl
```
Comment (hanzei) [inline source/configure/plugins-configuration-settings.rst:1426]: It is unclear to me which plugin these belong to and when an admin should modify these values. Can we clarify that?

File: source/collaborate/extend-mattermost-with-integrations.rst [hunk 4]
```diff
@@ -1,30 +1,24 @@
...
-* :doc:`Use the Mattermost Google Calendar plugin </collaborate/use-mattermost-google-calendar-plugin>` - Learn how to manage events using a two-way integration between Mattermost and Google Calendar.
\ No newline at end of file
+   `Visit the Mattermost Marketplace <https://mattermost.com/marketplace/>`__ to find open source, community-supported integrations to common developer tools like `CircleCI <https://mattermost.com/marketplace/circleci/>`__, `Opsgenie <https://mattermost.com/marketplace/opsgenie/>`__, `PagerDuty Notifier <https://mattermost.com/marketplace/pagerduty/>`__, and `Zendesk App <https://mattermost.com/marketplace/zendesk-app/>`__; productivity tools like `Autolink <https://mattermost.com/marketplace/autolink-plugin/>`__, `ToDo <https://mattermost.com/marketplace/todo/>`__, and `WelcomeBot <https://mattermost.com/marketplace/welcomebot-plugin/>`__; as well as social tools like `Memes <https://mattermost.com/marketplace/memes-plugin/>`__ and `GIFs <https://mattermost.com/marketplace/giphy-plugin/>`__ that are freely available for use and customization.
```
Comment (hanzei) [inline source/collaborate/extend-mattermost-with-integrations.rst:24]: I don't think it's wise to advertise the Zendesk App, given that we never used it in production and we are not sure it works.

**Review by hanzei — DISMISSED** [review]
Great work @cwarnermm :+1: 

Once this PR is merged, should we update the plugin metadata so the Marketplace entries point to the docs pages? Should we also remove the duplicate information from the respective READMEs?

File: source/guides/integrate-and-extend.rst (hunk 2 above)
Comment (cwarnermm) [reply source/guides/integrate-and-extend.rst:27]: AI Copilot isn't ready for release yet; will remove placeholders to it from this PR.

File: source/guides/integrate-and-extend.rst (hunk 1 above)
Comment (cwarnermm) [reply source/guides/integrate-and-extend.rst:33]: That's a really good point, @hanzei, and I agree with your classification of the Metrics plugin. Legal hold and channel export would also fall into this category. Perhaps the way forward is to include only the interoperability plugins here. Tool-based integrations would be documented as any other product feature. Will think on this more. Really appreciate your input here.

File: source/collaborate/extend-mattermost-with-integrations.rst (hunk 4 above)
Comment (cwarnermm) [reply source/collaborate/extend-mattermost-with-integrations.rst:24]: Thank you! I'll remove.

File: source/configure/plugins-configuration-settings.rst (hunk 3 above)
Comment (cwarnermm) [reply source/configure/plugins-configuration-settings.rst:1434]: The code highlighted is metadata used to return config settings in search results. 

I agree that the [Chimera OAuth proxy URL](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7030/configure/plugins-configuration-settings.html#chimera-oauth-proxy-url) docs don't specify which integrations this config affects, and we can definitely add this. Can you confirm which integrations utilize this config setting, please?

**Review by isacikgoz — APPROVED** [review]
Awesome, thanks @cwarnermm 💯

**Comment by hanzei** [comment]
I am unassinging myself per an offline discussion

File: source/integrate/github-interoperability.rst [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GitHub user tokens are AES-encrypted with an **At Rest Encryption Key** configured in Mattermost. Once encrypted, the tokens are saved in the ``PluginKeyValueStore`` table in your Mattermost database.
+
+Get help
+--------
+
+Mattermost customers can open a `Mattermost support case <https://mattermost.zendesk.com/hc/en-us/requests/new>`_. To report a bug, please open a GitHub issue against the `Mattermost GitHub plugin repository <https://github.com/mattermost/mattermost-plugin-github>`_.
+
+For questions, feedback, and assistance, join our pubic `Integrations and Apps channel <https://community.mattermost.com/core/channels/integrations>`_ on the `Mattermost Community Server <https://community.mattermost.com/>`_ for assistance, or join us on the  `Mattermost Discussion Forum <https://forum.mattermost.org/c/plugins>`_.
```
Comment (mickmister) [inline source/integrate/github-interoperability.rst:181]: This discussion forum link is taking me to a "page doesn't exist" page in the forum

File: source/integrate/github-interoperability.rst [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+A Mattermost system admin must perform the following steps in Mattermost.
+
+1. Confirm whether your Mattermost deployment has a ``github`` user account. If it exists, that account posts GitHub messages in channels by default, and the messages won't include a BOT tag. You can change this account behavior to include a BOT tag by using one of the following methods:
+
+ - Convert the user account to a bot using :ref:`mmctl user convert <manage/mmctl-command-line-tool:mmctl user convert>`.
+ - Change the existing ``github`` username to something else. A new bot account called ``github`` is created the Mattermost server is restarted when the :ref:`enable bot account creation <configure/integrations-configuration-settings:enable bot account creation>` configuration setting is enabled.
+
+.. note::
+
+    If you're using an older version of the GitHub integration (v0.9.0 and earlier), configure the username by going to **System Console > Plugins > GitHub**.
```
Comment (mickmister) [inline source/integrate/github-interoperability.rst:78]: I think we can remove this section

File: source/integrate/gitlab-interoperability.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+A Mattermost system admin must perform the following steps in Mattermost.
+
+Install the GitHub integration from the in-product App Marketplace:
+
+1. In Mattermost, from the Product menu |product-menu|, select **App Marketplace**.
+2. Search for or scroll to GitLab, and select **Install**.
+3. Once installed, select **Configure**. You're taken to the System Console.
+4. On the GitLab configuration page, enable and configure GitLab interoperability as follows, and then select **Save**:
+
+ - Enter the **GitLab URL**, **GitLab OAuth Client ID**, and **GitLab OAuth Client Secret** you obtained when `registering the OAuth app in GitLab <#register-an-oauth-app-in-GitLab>`__.
```
Comment (mickmister) [inline source/integrate/gitlab-interoperability.rst:58]: This link doesn't seem to make my browser scroll to the header

File: source/integrate/gitlab-interoperability.rst [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+  For GitLab versions prior to 1.2:
+
+  1. In GitLab, go to the project you want to subscribe to, and select **Settings > Integrations** in the sidebar.
+  2. Set the following values:
+
+    - **URL**: ``https://YOUR-MATTERMOST-URL.COM/plugins/com.github.manland.mattermost-plugin-gitlab/webhook``, replacing ``https://YOUR-MATTERMOST-URL.COM`` with your Mattermost URL. Ensure that you add ``/plugins/com.github.manland.mattermost-plugin-gitlab/webhook`` to the URL, or the webhook won't work.
+    - **Secret Token**: Copy the webhook secret you generated earlier.
+    - Select all the events in **Triggers**.
+    - Add the webhook.
```
Comment (mickmister) [inline source/integrate/gitlab-interoperability.rst:115]: I think we can remove this section

File: source/integrate/jira-interoperability.rst [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+    - **Jira Groups Allowed to Edit Jira Subscriptions**: (Applies to older Jira v2.4 or earlier deployments only) Specify the Jira groups allowed to edit Jira subscriptions as a comma-separated list of user group names. Leave blank to allow any Jira user the ability to create subscriptions. The user editing a subscription only needs to be a member of one of the listed groups.
+    - **Default Subscription Security Level to Empty**: Enable or disable default subscription security level. When enabled, subscriptions only include issues that have a security level assigned when a security level has been included as a filter.
+    - **Additional Help Text to be shown with Jira Help**: Define any additional help text to display when users run the ``/jira help`` slash command.
+    - **Hide issue descriptions and comments**: Show or hide issue descriptions and comments from subscription and webhook messages.
+    - **Enable slash command**: Enable or disble slash command autocompletion to guide users through available ``/jira`` slash commands.
+    - **Display Subscrition name in notifications**: Show or hide subscription name in notification messages posted to a channel.
```
Comment (mickmister) [inline source/integrate/jira-interoperability.rst:45]:
```suggestion
    - **Enable slash command**: Enable or disable slash command autocompletion to guide users through available ``/jira`` slash commands.
    - **Display Subscription name in notifications**: Show or hide subscription name in notification messages posted to a channel.
```

File: source/integrate/jira-interoperability.rst [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+
+6. Select **Save**.
+
+Legacy Jira webhooks
+^^^^^^^^^^^^^^^^^^^^
+
+If your organization's infrastructure is set up in such a way that your Mattermost instance can't connect to your Jira instance, you won't be able to subscribe Mattermost channels. You'll need to use legacy webhooks instead.
+
+1. To generate the webhook URL for a specific channel, run the  ``/jira webhook`` slash command, and use the URL output in the **Legacy Webhooks** section of the output.
+
```
Comment (mickmister) [inline source/integrate/jira-interoperability.rst:78]: I think this section could be de-emphasized, and put after the regular subscriptions section. Unrelated to the purpose of the PR though

File: source/integrate/jira-interoperability.rst [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+  - These subscription rules can specify the Jira Project, Event Type, Issue Type, and can filter out issues with certain values. 
+  - When a user is setting up a notification subscription, they'll only see the projects and issue types they have access to within Jira. If they can't see a project in Jira, it won't be displayed as an option for that particular user when they are trying to set up a subscription in Mattermost.
+  - If your organization's infrastructure is set up in such a way that your Mattermost instance can't connect to your Jira instance, channel subscriptions won't be avaialable. Instead, use `legacy Webhooks <#legacy-jira-webhooks>`__ instead to allow a Jira webhook to post to a specific channel.
+
+Manage notifications
+~~~~~~~~~~~~~~~~~~~~
+
+Jira notifications are messages sent to a Mattermost channel when a particular event occurs in Jira. They can managed as `channel subscriptions <#manage-channel-subscriptions>`__ in Mattermost, or managed as `webhooks <#configure-webhooks-in-Jira>`__ in Jira.
```
Comment (mickmister) [inline source/integrate/jira-interoperability.rst:133]: These two links don't seem to scroll on my browser

File: source/integrate/servicenow-interoperability.rst [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+3. Enter the name for your app and set the redirect URL to: ``https://<YOUR-MATTERMOST-URL>/plugins/mattermost-plugin-servicenow/api/v1/oauth2/complete``, replacing ``YOUR-MATTERMOST-URL`` with the Mattermost URL you want the ServiceNow events to post to, using lowercase characters.
+
+ .. note:: 
+
+  A client secret is generated automatically. Copy the secret and the Client ID. You'll need these values for the Mattermost configuration.
+
+Upload the update set in ServiceNow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Changing your ServiceNow instance to support subscriptions, record changes, and send change notifications to Mattermost is done by a Mattermost and ServiceNow system admin using an `update set <https://docs.servicenow.com/bundle/sandiego-application-development/page/build/system-update-sets/concept/system-update-sets.html>`_. Once created, you can download the update set from Mattermost and upload it into ServiceNow.
```
Comment (mickmister) [inline source/integrate/servicenow-interoperability.rst:32]: I think this docs link may have broken on SN's side

File: source/configure/plugins-configuration-settings.rst [hunk 13]
```diff
@@ -9,7 +9,6 @@ Self-hosted can manage the following configuration settings in **System Console
 - `Plugin Management <#plugin-management>`__
 - `Apps <#apps>`__
 - `Calls <#calls>`__
-- `Google Calendar <#google-calendar>`__
 - `MS Teams <#ms-teams>`__
```
Comment (mickmister) [inline source/configure/plugins-configuration-settings.rst:12]: The MS Teams section is pretty big on this page. Maybe those settings could be put in another page like other integration plugins on the page?

**Review by mickmister — APPROVED** [review]
Amazing work @cwarnermm!! Crazy how much attention to detail this took, migrating markdown to rst alone is quite the task, and organizing it all this way with everything working 👏 

LGTM 🚀

File: source/integrate/zoom-interoperability.rst [hunk 14]
```diff
@@ -0,0 +1,153 @@
+Connect Zoom to Mattermost
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/integrate/zoom-interoperability.rst:1]: aha! Great callout, @mickmister - thank you! I agree that this PR will impact the newly-formed integration docs via docs.mm.com.

File: source/integrate/zoom-interoperability.rst [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+===========================
+
+.. include:: ../_static/badges/allplans-cloud-selfhosted.rst
+  :start-after: :nosearch:
+
+.. |product-menu| image:: ../images/products_E82F.svg
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/integrate/zoom-interoperability.rst:7]: Those filenames come directly from the [mattermost/compass-icons](https://github.com/mattermost/compass-icons/tree/master/svgs) GH repository. I've matched the filename convention in the hopes of finding a way to automate the inclusion of these icons in docs some day.

File: source/guides/integrate-and-extend.rst [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+-----------------------------------------------
+
+Your Mattermost deployment comes with the following integrations you can configure and use:
+
+* :doc:`GitHub interoperability </integrate/github-interoperability>` - Connect your GitHub instance to your Mattermost instance.
+* :doc:`GitLab interoperability </integrate/gitlab-interoperability>` - Connect your GitLab instance to your Mattermost instance.
+* :doc:`Jira interoperability </integrate/jira-interoperability>` - Connect your Jira instance to your Mattermost instance.
+* :doc:`Microsoft Teams interoperability </integrate/ms-teams-interoperability>` - Connect your operational workflows through the Mattermost app while staying connected to an all-employee platform, all in one place.
+* :doc:`ServiceNow interoperability </integrate/servicenow-interoperability>` - Connect your ServiceNow to your Mattermost instance.
+* :doc:`Zoom interoperability </integrate/zoom-interoperability>` - Learn how to connect your Zoom instance to your Mattermost instance.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/guides/integrate-and-extend.rst:28]: A most excellent point about consistency. Thank you! I'll update.

File: source/guides/integrate-and-extend.rst [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+  :titlesonly:
+
+    GitHub interoperability </integrate/github-interoperability>
+    GitLab interoperability </integrate/gitlab-interoperability>
+    Jira interoperability </integrate/jira-interoperability>
+    Microsoft Teams interoperability </integrate/ms-teams-interoperability>
+    ServiceNow interoperability </integrate/servicenow-interoperability>
+    Zoom interoperability </integrate/zoom-interoperability>
+
+Mattermost offers many ways to add product functionality, custom-tailor your workflows, and connect Mattermost to your mission-critical tools.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/guides/integrate-and-extend.rst:16]: Agreed that the language here isn't really adding value. Will revisit. Thank you!

File: source/integrate/jira-interoperability.rst (hunk 10 above)
Comment (cwarnermm) [reply source/integrate/jira-interoperability.rst:78]: A very good piece of feedback that I'll action. Thanks!

File: source/integrate/jira-interoperability.rst (hunk 11 above)
Comment (cwarnermm) [reply source/integrate/jira-interoperability.rst:133]: I'll investigate and correct.

File: source/integrate/servicenow-interoperability.rst (hunk 12 above)
Comment (cwarnermm) [reply source/integrate/servicenow-interoperability.rst:32]: Thanks for this heads-up! I'll investigate and fix.

File: source/configure/plugins-configuration-settings.rst (hunk 13 above)
Comment (cwarnermm) [reply source/configure/plugins-configuration-settings.rst:12]: That's exactly the longer-term plan that will happen in a future release.

File: source/guides/integrate-and-extend.rst (hunk 1 above)
Comment (cwarnermm) [reply source/guides/integrate-and-extend.rst]: Features built as plugins are documented like features; 3rd party interoperability integrations are collected together, in the docs, under Integrate & Extend from the landing page.

I agree that we'll want to update the plugin metadata so the Marketplace entries point to the docs pages -- both the in-product Marketplace as well as the external one via mm.com/marketplace. These tasks will be completed outside of this PR.

File: source/integrate/jira-interoperability.rst (hunk 11 above)
Comment (cwarnermm) [reply source/integrate/jira-interoperability.rst:133]: Fixed

**Pull request merged**